src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -31,27 +35,87 @@ public class Engine : IPowerFxEngine
         /// <param name="powerFxConfig"></param>
         public Engine(PowerFxConfig powerFxConfig)
         {
-            powerFxConfig.Lock();
             Config = powerFxConfig;
         }
 
+        // All functions that powerfx core knows about. 
+        // Derived engines may only support a subset of these builtins, 
+        // and they may add their own custom ones. 
+        private static readonly ReadOnlySymbolTable _allBuiltinCoreFunctions = NewDefault(Core.Texl.BuiltinFunctionsCore.BuiltinFunctionsLibrary);
+
+        /// <summary>
+        /// Builtin functions supported by this engine. 
+        /// </summary>
+        public ReadOnlySymbolTable SupportedFunctions { get; set; } = _allBuiltinCoreFunctions;
```

Comment:
>set [](http://example.com/codeflow?start=61&length=3)

Should this be either protected or internal set?

Reply:
protected set is good. 
And then engines that can be flexible here can expose options to update it.  (say if you wanted to remove a function from the interpreter)